src/main/java/com/gachigang/ontherun/config/security/jwt/JwtAuthenticationFilter.java

```diff
@@ -0,0 +1,70 @@
+package com.gachigang.ontherun.config.security.jwt;
+
+import com.gachigang.ontherun.persistence.repository.TokenRepository;
+import com.gachigang.ontherun.service.JwtService;
+import jakarta.servlet.FilterChain;
+import jakarta.servlet.ServletException;
+import jakarta.servlet.http.HttpServletRequest;
+import jakarta.servlet.http.HttpServletResponse;
+import lombok.RequiredArgsConstructor;
+import org.springframework.http.HttpHeaders;
+import org.springframework.lang.NonNull;
+import org.springframework.security.authentication.UsernamePasswordAuthenticationToken;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.security.core.userdetails.UserDetailsService;
+import org.springframework.security.web.authentication.WebAuthenticationDetailsSource;
+import org.springframework.stereotype.Component;
+import org.springframework.web.filter.OncePerRequestFilter;
+
+import java.io.IOException;
+
+import static com.gachigang.ontherun.common.ApplicationConstants.Security.TOKEN_PREFIX;
+
+@Component
+@RequiredArgsConstructor
+public class JwtAuthenticationFilter extends OncePerRequestFilter {
+
+    private final JwtService jwtService;
+    private final UserDetailsService userDetailsService;
+    private final TokenRepository tokenRepository;
+
+    @Override
+    protected void doFilterInternal(
+            @NonNull HttpServletRequest request,
+            @NonNull HttpServletResponse response,
+            @NonNull FilterChain filterChain
+    ) throws ServletException, IOException {
+        if (request.getServletPath().contains("/api/auth")) {
+            filterChain.doFilter(request, response);
+            return;
+        }
+        final String authHeader = request.getHeader(HttpHeaders.AUTHORIZATION);
+        final String jwt;
+        final String userEmail;
+        if (authHeader == null ||!authHeader.startsWith(TOKEN_PREFIX)) {
+            filterChain.doFilter(request, response);
+            return;
+        }
+        jwt = authHeader.substring(TOKEN_PREFIX.length());
+        userEmail = jwtService.extractUsername(jwt);
```

Comment:
If access token will expire you will get JwtExpireException here and it will be printed with stack trace in console even if refresh token avaliable and not expired -- need to find oud a way for handle it

Reply:
I`m going to look for a better solution.